src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c

```diff
@@ -21,23 +36,160 @@ const ActorInit Bg_Jya_Zurerukabe_InitVars = {
+static s16 D_8089B9F0[4] = { 943, 1043, 1243, 1343 };
+
+static s16 D_8089B9F8[4] = { -1, 1, -1, 1 };
+
+static s16 D_8089BA00[4] = { 48, 48, 36, 36 };
+
+static f32 D_8089BA08[4] = { 8.0f, 8.0f, 10.0f, 10.0f };
+
+static s16 D_8089BA18[6][2] = {
+    { 0x0388, 0x0395 }, { 0x03EA, 0x03FF }, { 0x0454, 0x0467 },
+    { 0x04B4, 0x04C1 }, { 0x0518, 0x0528 }, { 0x0581, 0x0590 },
+};
+
+static s16 D_8089BA30[6] = {
+    0, 0, 1, 2, 2, 3,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 1200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+extern Gfx D_06012340[];
+extern UNK_PTR D_06012508;
+
+void func_8089B440(BgJyaZurerukabe* this, GlobalContext* globalCtx, void* arg2, s32 flags) {
+    s32 pad;
+    s32 localC = 0;
+    s32 pad2;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, flags);
+    DynaPolyInfo_Alloc(arg2, &localC);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, localC);
+
+    if (this->dyna.dynaPolyId == 0x32) {
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_jya_zurerukabe.c", 194,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void func_8089B4C8(BgJyaZurerukabe* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((player->stateFlags1 == 0x200000) && (player->actor.wallPoly != NULL)) {
+        s32 i;
+
+        for (i = 0; i < ARRAY_COUNT(D_8089BA18); i++) {
+            f32 posY = player->actor.posRot.pos.y;
+            if ((posY >= D_8089BA18[i][0]) && (posY <= D_8089BA18[i][1])) {
+                break;
+            }
+        }
+
+        switch (i) {
+            case 0:
+            case 2:
+            case 3:
+            case 5:
+                if (fabsf(D_8089B9C0[D_8089BA30[i]]) > 1.0f) {
+                    func_8002F6D4(globalCtx, &this->dyna.actor, 1.5f, this->dyna.actor.shape.rot.y, 0.0f, 0);
+                }
+                break;
+            case 1:
+            case 4:
+                if (fabsf(D_8089B9C0[D_8089BA30[i]] - D_8089B9C0[D_8089BA30[i + 1]]) > 1.0f) {
+                    func_8002F6D4(globalCtx, &this->dyna.actor, 1.5f, this->dyna.actor.shape.rot.y, 0.0f, 0);
+                }
+                break;
+        }
+    }
+}
+
+void BgJyaZurerukabe_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaZurerukabe* this = THIS;
+    s32 i;
+
+    func_8089B440(this, globalCtx, &D_06012508, DPM_UNK);
+    Actor_ProcessInitChain(thisx, sInitChain);
+
+    for (i = 0; i < ARRAY_COUNT(D_8089B9F0); i++) {
+        if (fabsf(D_8089B9F0[i] - this->dyna.actor.initPosRot.pos.y) < 1.0f) {
+            this->unk_168 = i;
+            break;
+        }
+    }
+
+    if (D_8089B9F8 == &D_8089B9F0[i]) {
```

Comment:
```suggestion
    if (i == ARRAY_COUNT(D_8089B9F0)) {
```

Reply:
Yeah I agree it'd look a lot better, but I don't think it'll work..I'll try though

Reply:
...nevermind! worked :)